src/modules/uORB/uORBDeviceMaster.cpp

```diff
@@ -454,7 +455,11 @@ uORB::DeviceNode *uORB::DeviceMaster::getDeviceNode(const char *nodepath)
 
 uORB::DeviceNode *uORB::DeviceMaster::getDeviceNode(const struct orb_metadata *meta, const uint8_t instance)
 {
-	if (meta == nullptr) {
+	if ((meta == nullptr) || (instance > ORB_MULTI_MAX_INSTANCES - 1)) {
+		return nullptr;
+	}
+
+	if ((static_cast<ORB_ID>(meta->o_id) != ORB_ID::INVALID) && !_node_exists[instance][meta->o_id]) {
```

Comment:
`static_cast<ORB_ID>(meta->o_id) != ORB_ID::INVALID` is not strictly necessary

Reply:
Yes, just being paranoid. With some special handling for logger I don't think we even need `ORB_ID::INVALID` for all other usage. Something to come back to later.

Reply:
Consolidated in `uORB::DeviceMaster::deviceNodeExists()`.